### ISSUE TITLE
IPv6 Nameservers are not purged when changing wifi networks

### ISSUE THREAD

**Issue by mweinelt** [systemd/systemd#3870]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)
### systemd version the issue has been seen with

systemd 231
with RDNSS support
### Used distribution

Linux 4.6.4-1-ARCH 
wpa_supplicant 1:2.5-3
### In case of bug report: Expected behaviour you didn't see

IPv6 nameservers should be removed from `/etc/resolv.conf` when changing the wifi network. This works for IPv4.
### In case of bug report: Unexpected behaviour you saw

IPv6 nameservers remained in `/etc/resolv.conf` after change of wifi network. IPv4 nameservers were purged.
### In case of bug report: Steps to reproduce the problem
1. Connect to IPv6 capable wifi network 1
2. Receive IPv6 nameservers via RDNSS option in RADV
3. Connect to wifi network 2
4. Receive IPv6 nameservers via RDNSS again
5. Nameservers are now a mix of networks 1 and 2 nameservers

**Comment by poettering** [comment]
hmm, interesting. What does "networkctl status -a" show about the DNS servers? And what does "systemd-resolve --status" show?

Background: we need to know if networkd's idea of the DNS config is borked, or resolved's idea.

**Comment by mweinelt** [comment]
Network 1

```
● 3: wl0
       Link File: /etc/systemd/network/10-wl0.link
    Network File: /etc/systemd/network/20-wl0.network
            Type: wlan
           State: routable (configured)
            Path: pci-0000:03:00.0
          Driver: iwlwifi
          Vendor: Intel Corporation
           Model: Wireless 7260 (Dual Band Wireless-AC 7260)
      HW Address: 72:c1:2c:e2:0d:42
         Address: 172.23.97.196
                  2003:87:2f16:ec10:70c1:2cff:fee2:d42
                  fd42:ccc:da:1:70c1:2cff:fee2:d42
                  fe80::70c1:2cff:fee2:d42
         Gateway: 172.23.97.254 (Ubiquiti Networks Inc.)
                  fe80::de9f:dbff:fe80:1e8 (Ubiquiti Networks Inc.)
             DNS: 8.8.8.8
                  8.8.4.4
                  2003:180:2:8100::53
                  2003:180:2:8000::53
  Search Domains: trollhoehle.net

```

```
Link 3 (wl0)
      Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
       LLMNR setting: yes
MulticastDNS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
         DNS Servers: 8.8.8.8
                      8.8.4.4
                      2003:180:2:8000::53
                      2003:180:2:8100::53
          DNS Domain: trollhoehle.net
```

switching to Network 2

```
● 3: wl0
       Link File: /etc/systemd/network/10-wl0.link
    Network File: /etc/systemd/network/20-wl0.network
            Type: wlan
           State: routable (configured)
            Path: pci-0000:03:00.0
          Driver: iwlwifi
          Vendor: Intel Corporation
           Model: Wireless 7260 (Dual Band Wireless-AC 7260)
      HW Address: b6:65:cc:a6:7f:d9
         Address: 10.223.21.173
                  2001:67c:2ed8:6100:b465:ccff:fea6:7fd9
                  fd01:67c:2ed8:6100:b465:ccff:fea6:7fd9
                  fe80::b465:ccff:fea6:7fd9
         Gateway: 10.223.0.5
                  fe80::d8ff:61ff:fe00:104
                  fe80::d8ff:61ff:fe00:204
                  fe80::d8ff:61ff:fe00:504
             DNS: 10.223.254.55
                  10.223.254.56
                  2003:180:2:8100::53
                  fd01:67c:2ed8:a::56:1
                  2003:180:2:8000::53
                  fd01:67c:2ed8:a::55:1
  Search Domains: ffda.io
```

```
Link 3 (wl0)
      Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
       LLMNR setting: yes
MulticastDNS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
         DNS Servers: 10.223.254.55
                      10.223.254.56
                      2003:180:2:8100::53
                      fd01:67c:2ed8:a::56:1
                      2003:180:2:8000::53
                      fd01:67c:2ed8:a::55:1
          DNS Domain: ffda.io
```

**Comment by mweinelt** [comment]
Is more info needed?

**Comment by poettering** [comment]
I figure we have to flush out the RA data we collected whenever the link goes down temporarily in networkd.

**Comment by poettering** [comment]
fix waiting in #4481
